Library/Homebrew/extend/os/mac/software_spec.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+class SoftwareSpec
+  def uses_from_macos(deps, **args)
+    depends_on(deps) if add_mac_dependency?(args)
+  end
+
+  private
+
+  def add_mac_dependency?(args)
+    args.each { |key, version| args[key] = OS::Mac::Version.from_symbol(version) }
+
+    return false if args[:after] && OS::Mac.version < args[:after]
+
+    return false if args[:before] && OS::Mac.version >= args[:before]
+
+    true
```

Comment:
> From my understanding, uses_from_macos "foo" (i.e. without version args) should not add foo as dependency on macOS. However, the current logic does the opposite.

```suggestion
    !args.empty?
```

Reply:
Well, actually i think it makes sense to have this behaviour only on Linux, since it **uses from macos**. Without any version specification it seems that the method uses from macos unconditionally, so:

On MacOS
```rb
uses_from_macos "foo"                    # always add foo as dep in macos
uses_from_macos "foo", after: :sierra    # add foo as dep only after macos sierra
```
On Linux
```rb
uses_from_macos "foo" # Does nothing, since we're adding a macos dependency
```

What do you think? @sjackman @xu-cheng

Reply:
In fact it's the exact opposite of that. `uses_from_macos "foo", after: :sierra` means that on macOS Sierra and later we use `foo` provided by macOS operating system, and so we do *not* need to to use the brewed dependency. Before macOS Sierra we need to install the brewed dependency. On Linux we always install the dependency.

On macOS
```ruby
uses_from_macos "foo"                    # Never add foo as dep in macos
uses_from_macos "foo", after: :sierra    # add foo as dep only before macos sierra
```
On Linux
```ruby
uses_from_macos "foo" # Always add foo as dep on Linux
```